bin/weewx/drivers/vantage.py

```diff
@@ -344,33 +353,62 @@ def queued_bytes(self):
             self.socket.settimeout(self.timeout)
         return length
 
-    def read(self, chars=1):
+    def read(self, chars=1, flush=False):
         """Read bytes from WeatherLinkIP"""
         import socket
         _buffer = ''
+        _recv = ''
         _remaining = chars
         while _remaining:
             _N = min(4096, _remaining)
             try:
                 _recv = self.socket.recv(_N)
-            except (socket.timeout, socket.error), ex:
+            except socket.timeout:
```

Comment:
I work by practising, Tom. First I add debug lines at crucial points in the original source, then notice things which go wrong and try to find a solution for that. But my coding skill are very poor (my problem). Maybe I should describe the problems I noticed in the behaviour of the original driver, suggest solutions for that and let somebody with (enough) program skills 'do the job'. Is that an idea?